firebase.d.ts

```diff
@@ -786,6 +786,8 @@ export function login(options: LoginOptions): Promise<User>;
 
 export function reauthenticate(options: ReauthenticateOptions): Promise<any>;
 
+export function reload(): Promise<void>;
```

Comment:
I may change this to `reloadUser` to make it more clear, but I'll make up my mind after merging this awesome PR.

Reply:
I second the idea of renaming to make the function more clear 👍🏼